test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -197,6 +197,16 @@ void UcxContext::disconnect(UcxConnection* conn) _GLIBCXX_NOTHROW
     }
 }
 
+void UcxContext::on_disconnect(UcxConnection* conn) _GLIBCXX_NOTHROW
+{
+    verbose_os() << "closing connection " << conn << " with id = " << conn->id()
+                 << std::endl;
+    remove_connection(conn);
+    if (!conn->on_disconnect()) {
+        _closing_conns.push_back(conn);
+    }
```

Comment:
instead of doing these in UcxContext "base" implementation, better keep the base implementation empty (or even abstract) and do these cleanup before/after calling on_disconnect()

Reply:
why?